### ISSUE TITLE
    if (bbox_intersection_area(ba, bb) / bbox_area(ba)) > 0.8: ZeroDivisionError: float division by zero

### ISSUE THREAD

**Issue by arjungandeeva** [camelot-dev/camelot#495]
I'm encountering a ZeroDivisionError: float division by zero error in camelot-py when using the functions bbox_intersection_area and bbox_area. This error occurs under certain conditions, likely when the bounding box area (ba) is zero.

**Comment by cktse** [comment]
I did a quick fix/hack to circumvent the error by skipping over the area check if ba is singular (area is zero):

/Users/cktse/.pyenv/versions/3.11.3/lib/python3.11/site-packages/camelot/utils.py: Line 375:

```
            if bbox_area(ba) > 0 and bbox_intersect(ba, bb):
                # if the intersection is larger than 80% of ba's size, we keep the longest
                if (bbox_intersection_area(ba, bb) / bbox_area(ba)) > 0.8:
                    if bbox_longer(bb, ba):
                        rest.discard(ba)
```

**Comment by bosd** [comment]
Hey!

As https://github.com/camelot-dev/camelot/issues/343, we try to build a maintained fork at [pypdf_table_extraction](https://github.com/py-pdf/pypdf_table_extraction).

Do you want to check that code and open an issue / PR thereto include this fix?

**Comment by bosd** [comment]
Thanks for checking 👍

**Comment by cktse** [comment]
Great to see camelot lives on!

BTW is this fork going to be packaged on pip under a separate name? Think the current package is stale from the main branch.

**Comment by bosd** [comment]
> BTW is this fork going to be packaged on pip under a separate name? Think the current package is stale from the main branch.

Yes, it is published here https://pypi.org/project/pypdf-table-extraction/

We're currently working on a new release, bymerging the open pr's from this repo, and rebranding the package.
